src/media-segment-request.js

```diff
@@ -102,31 +96,29 @@ const getProgressStats = (progressEvent) => {
  * @param {Object} request -  the XHR request that possibly generated the error
  */
 const handleErrors = (error, request) => {
-  const response = request.response;
-
-  if (!request.aborted && error) {
+  if (request.timedout) {
```

Comment:
The `error` parameter can be set for a variety of reasons. The only time we should consider it an actual error is when neither `timedout` nor `aborted` are set.

The issue was that a `timeout` was being interpreted as an error because the `error` parameter was set and the request wasn't aborted.

In the unit tests, Sinon's FakeXHR *was* setting the `aborted` property on timeouts avoiding this condition but the browser's native XHR does not set that property.